connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -1466,560 +1032,421 @@ public void testRestartConnectorAndTasksUnknownConnector() throws Exception {
+        doReturn(Optional.of(restartPlan)).when(herder).buildRestartPlan(restartRequest);
 
-        PowerMock.replayAll();
         herder.assignment = ExtendedAssignment.empty();
         herder.doRestartConnectorAndTasks(restartRequest);
-        PowerMock.verifyAll();
+
+        verifyNoMoreInteractions(worker, member, configBackingStore, statusBackingStore);
     }
 
     @Test
     public void testDoRestartConnectorAndTasksOnlyConnector() {
         ConnectorTaskId taskId = new ConnectorTaskId(CONN1, 0);
         RestartRequest restartRequest = new RestartRequest(CONN1, false, true);
-        RestartPlan restartPlan = PowerMock.createMock(RestartPlan.class);
-        EasyMock.expect(restartPlan.shouldRestartConnector()).andReturn(true).anyTimes();
-        EasyMock.expect(restartPlan.shouldRestartTasks()).andReturn(true).anyTimes();
-        EasyMock.expect(restartPlan.taskIdsToRestart()).andReturn(Collections.singletonList(taskId)).anyTimes();
+        RestartPlan restartPlan = mock(RestartPlan.class);
+        when(restartPlan.shouldRestartConnector()).thenReturn(true);
+        when(restartPlan.taskIdsToRestart()).thenReturn(Collections.singletonList(taskId));
 
-        EasyMock.expect(herder.buildRestartPlan(restartRequest)).andReturn(Optional.of(restartPlan)).anyTimes();
+        doReturn(Optional.of(restartPlan)).when(herder).buildRestartPlan(restartRequest);
 
-        herder.assignment = PowerMock.createMock(ExtendedAssignment.class);
-        EasyMock.expect(herder.assignment.connectors()).andReturn(Collections.singletonList(CONN1)).anyTimes();
-        EasyMock.expect(herder.assignment.tasks()).andReturn(Collections.emptyList()).anyTimes();
+        herder.assignment = mock(ExtendedAssignment.class);
+        when(herder.assignment.connectors()).thenReturn(Collections.singletonList(CONN1));
+        when(herder.assignment.tasks()).thenReturn(Collections.emptyList());
 
-        worker.stopAndAwaitConnector(CONN1);
-        PowerMock.expectLastCall();
+        doNothing().when(worker).stopAndAwaitConnector(CONN1);
 
-        Capture<Callback<TargetState>>  stateCallback = newCapture();
-        worker.startConnector(EasyMock.eq(CONN1), EasyMock.anyObject(), EasyMock.anyObject(),
-                EasyMock.eq(herder), EasyMock.anyObject(TargetState.class), capture(stateCallback));
-        PowerMock.expectLastCall().andAnswer(() -> {
+        ArgumentCaptor<Callback<TargetState>>  stateCallback = ArgumentCaptor.forClass(Callback.class);
+        doAnswer(invocation -> {
             stateCallback.getValue().onCompletion(null, TargetState.STARTED);
             return true;
-        });
-        member.wakeup();
-        PowerMock.expectLastCall();
+        }).when(worker).startConnector(eq(CONN1), any(), any(), eq(herder), any(), stateCallback.capture());
+        doNothing().when(member).wakeup();
+        doNothing().when(statusBackingStore).put(any(ConnectorStatus.class));
```

Comment:
This is a little weak in terms of coverage; we could be failing the connector here instead of marking it as restarting.

There are a handful of ways to address this. One could be to set up a mock expectation with the exact status we expect to see reported to the status store:
```suggestion
        ConnectorStatus status = new ConnectorStatus(CONN1, AbstractStatus.State.RESTARTING, WORKER_ID, 0);
        doNothing().when(statusBackingStore).put(status);
```

Reply:
> One could be to set up a mock expectation with the exact status we expect to see reported to the status store

Makes sense, I've made this change here and in the subsequent tests as well. I guess I just put in the lowest impedance change initially since it's additional coverage (but was required to satisfy the new stricter verifications) 😅